### PR TITLE
Add option to change the app's language

### LINKED ISSUE #3329: Can't change language.
**Summary:**

I'm going to settings and there is only an option for changing the default description language.
The default app description is been set by the device's default language and cannot be changed.

**Steps to reproduce:** 

We should  add an option where user could also choose for the default app language.
So after the User choose an other language , the program should re-choose  the right resource bundle and re-load the program's default String Values .

**Would you like to work on the issue?**

I want to give it a try. I'm new in contribution, actually this would be my first, and i would need help in solving it and sure some directions. 
e.x. In which Class should i contribute? , should i reload the hole program or i could change the string values one by one ?.....

### PR DESCRIPTION
**Pull request by 6point022**
**Description (required)**

I added a preference which allows a user to change the app's language by going to the Settings.

Fixes #3329 

**What changes did you make and why?**

I added another `ListPreference` in the `preferences.xml` file, and in the `SettingsFragment.java` I set it up with the array items and a listener. Then I implemented the `setAppLanguage` method in the `MainActivity` which changes the resource configuration, and hence, the preferred language's `strings.xml` file is used. This method is called every time the app opens up which allows us to change the language of the app to the user's last saved preference.

**Tests performed (required)**

Tested betaDebug on Redmi Note 5 Pro with API level 28.

**Screenshots showing what changed (optional - for UI changes)**

The following screenshots were taken after changing the app's language to Hindi.

image

image

image

image

---

_This is my first PR to this project. I tried to make sure I follow all the contribution guidelines mentioned.  But I may have made some mistakes (inevitable), and I would really appreciate it if someone could point those out. I'll try my best to fix them._

### CODE REVIEW

**Comment by 6point022** [comment]
@neslihanturan I have made the requested changes. Please review.

File: app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java [hunk 1]
```diff
@@ -80,6 +86,7 @@
 
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setAppLanguage(defaultKvStore.getString(Prefs.KEY_APP_LANGUAGE_VALUE));
```
[the remark this answers is not in the document]
Comment (6point022) [reply app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java]: @neslihanturan If I don't set the language on app startup, it switches to the default language upon reopening the app. I tried looking for other ways to solve this, but most of the solutions were about setting the language when the app starts.

[the remark this answers is not in the document]
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java]: You can use SharedPreferences. We use JsonKvStore for this purpose, you can search this line to get idea about implementation in our code, and do the same.
```
    @Inject @Named("default_preferences") JsonKvStore kvStore;
```

**Comment by misaochan** [comment]
Closing this due to community decision at #3329 . Just wanted to note that this is a fantastic PR that fits our code quality requirements, we just weren't able to merge it for other reasons.

**Pull request closed without merging**